From 609199b208680991b8e1ec810e88cbcc8cd63ac2 Mon Sep 17 00:00:00 2001
From: Shilin Victor <chrono.monochrome@gmail.com>
Date: Mon, 20 Aug 2018 22:57:43 +0300
Subject: [PATCH 043/296] Fix NotificationListenerService monkey null pointer
 issues

Monkey test will causes null pointer for array.

Add a null pointer protection.
---
 core/java/android/service/notification/NotificationListenerService.java | 1 +
 1 file changed, 1 insertion(+)

diff --git a/core/java/android/service/notification/NotificationListenerService.java b/core/java/android/service/notification/NotificationListenerService.java
index a5223fd8acd..e7a3f1bd521 100644
--- a/core/java/android/service/notification/NotificationListenerService.java
+++ b/core/java/android/service/notification/NotificationListenerService.java
@@ -857,6 +857,7 @@ public abstract class NotificationListenerService extends Service {
 
     private StatusBarNotification[] cleanUpNotificationList(
             ParceledListSlice<StatusBarNotification> parceledList) {
+        if (parceledList == null) return null;
         List<StatusBarNotification> list = parceledList.getList();
         ArrayList<StatusBarNotification> corruptNotifications = null;
         int N = list.size();
-- 
2.11.0

